toolkit/tools/imagegen/configuration/configuration.go

```diff
@@ -208,12 +208,12 @@ func convertRawBinariesPath(baseDirPath string, diskConfig *Disk) {
 }
 
 func convertAdditionalFilesPath(baseDirPath string, systemConfig *SystemConfig) {
+	absAdditionalFiles := make(map[string]string)
 	for localFilePath, targetFilePath := range systemConfig.AdditionalFiles {
-		delete(systemConfig.AdditionalFiles, localFilePath)
```

Comment:
Both of these are good catches. I'm actually surprised why we decided to delete any entries in these loops to be honest, since we only care about updating the values. I'm curious if this would also work, if we simply removed the `delete...` line, since then we wouldn't be hitting that "creating entries during iteration" scenario? At least that's my hope.:)